zone/client_packet.cpp

```diff
@@ -526,6 +538,9 @@ void Client::CompleteConnect()
 	/* Sets GM Flag if needed & Sends Petition Queue */
 	UpdateAdmin(false);
 
+	// Task Packets
+	LoadClientTaskState();
```

Comment:
We'll need to keep an eye on this, I think you were right and there's still some potential for races even with this fix

Reply:
Ack